monkey/monkey_island/cc/services/attack/attack_config.py

```diff
@@ -0,0 +1,162 @@
+import logging
+from dpath import util
+from monkey_island.cc.database import mongo
+from monkey_island.cc.services.attack.attack_schema import SCHEMA
+from monkey_island.cc.services.config import ConfigService
+
+__author__ = "VakarisZ"
+
+logger = logging.getLogger(__name__)
+
```

Comment:
How does it become more readable by having `@staticmethod`, decorators and other useless code?

Reply:
The `@staticmethod` is the only real addition so it's pretty easy.
The advantage of this pattern is that when you import the class and use it, it is more readable.
In this specific case, this is how using it now looks:
```
from monkey_island.cc.services.attack.attack_config import get_config
...
get_config()
```
This is how it would look like with static class:
```
from monkey_island.cc.services.attack.attack_config import AttackConfig
...
AttackConfig.get_config()
```

Now the reader understands `get_config` refers to the attack config and not the regular config.

As far as I know and was able to find, this pattern is used. not considered bad practice yet not best practice. A matter of taste.

Since this is not a bad pattern, and all our services are using it, being consistent in style for the project will make the project as a whole more readable.

Reply:
I'm not saying it's hard to change, I don't like it syntactically. Given how much we try to avoid code duplication, `@staticmethod` and Class definition with `__init__(self): pass` would add ~ 17 unused lines of code for this file. Also, I don't see how 
`from monkey_island.cc.services.attack.attack_config import AttackConfig`
`...`
`AttackConfig.get_config()`
is better than what we have now:
`import monkey_island.cc.services.attack.attack_config as attack_config`
`...`
`attack_config.get_config()`
I can refactor it without a problem the question is if this approach (making static classes) should be used in future development. I would say no.

Reply:
The advantage of the first is that it forces whoever uses AttackConfig to do `AttackConfig.get_config()`
On the second option the developer has a choice of doing
```
from monkey_island.cc.services.attack.attack_config import get_config
...
get_config()
```

It really doesn't seem crucial to me which option we choose, and I don't see an actual problem with the extra c'tor and `@staticmehod` decorators.
@danielguardicore - your opinion on this so we can end this discussion